code/xcvm/cosmwasm/contracts/gateway/src/error.rs

```diff
@@ -38,4 +38,16 @@ pub enum ContractError {
 	UnknownChannel,
 	#[error("The asset is already registered.")]
 	AlreadyRegistered,
+	#[error("Route not found.")]
+	RouteNotFound,
+	#[error("{0:?}")]
+	Bech32(bech32_no_std::Error),
+	#[error("{0:?}")]
+	Serde(#[from] serde_json_wasm::ser::Error),
+}
+
+impl From<bech32_no_std::Error> for ContractError {
+	fn from(value: bech32_no_std::Error) -> Self {
+		Self::Bech32(value)
+	}
 }
```

Comment:
```suggestion
	#[error("{0}")]
	Bech32(#[from] bech32_no_std::Error),
	#[error("{0}")]
	Serde(#[from] serde_json_wasm::ser::Error),
}
```

Reply:
hm, for some reason bech32 does not work like that